src/modules/apps/exec.ts

```diff
@@ -0,0 +1,53 @@
+import * as WebSocket from 'ws'
+import * as url from 'url'
+import { getAccount, getToken, getWorkspace } from '../../conf'
+import { getManifest } from '../../manifest'
+import { region } from '../../env'
+import { toMajorRange } from '../../locator'
+const EOT = '\x04'
+export default async (command: string, options) => {
+  const manifest = await getManifest()
+  const { name, vendor, version, builders } = manifest
+  const { dotnet, node, 'service-js': serviceJs } = builders
+  if (!dotnet && !node && !serviceJs) {
+    return
+  }
+  const majorRange = toMajorRange(version)
+  const host = `${name}.${vendor}.${region()}.vtex.io`
+  const path = `/${getAccount()}/${getWorkspace()}/_exec`
+  const clientOptions = {
+    headers: {
+      Authorization: getToken(),
+      Host: host,
+      'X-Vtex-Runtime-Api': 'true',
+    },
+  }
+  var urlObject = {
+    protocol: 'ws',
+    hostname: host,
+    pathname: path,
+    query: {
+      __v: majorRange,
+      params: command?.split(' '),
+      interactive: options.i || options.interactive,
+    },
+  }
+  const formattedUrl = url.format(urlObject)
+  const ws = new WebSocket(formattedUrl, clientOptions)
+  ws.on('open', function open() {
+    process.stdin.resume()
+    process.stdin.setEncoding('utf8')
+    process.stdin.on('data', function(chunk) {
+      ws.send(chunk)
+    })
+    process.stdin.on('end', function() {
+      ws.send(EOT)
+    })
+  })
+  ws.on('close', function close() {
+    process.exit(0)
+  })
+  ws.on('message', function incoming(data) {
+    process.stdout.write(data.toString())
+  })
+}
```

Comment:
How about using the `WebSocket` [Streams API](https://www.npmjs.com/package/ws#use-the-nodejs-streams-api)?
I think this could be:
```ts
const ws = new WebSocket('formattedUrl', clientOptions)
const wsDuplexStream = WebSocket.createWebSocketStream(ws, { encoding: 'utf8' });
process.stdin.pipe(wsDuplexStream)
wsDuplexStream.pipe(process.stdout)
```
I think it would needed to be tested if `EOT` is necessary in this case, otherwise it could be:
```ts
const ws = new WebSocket('formattedUrl', clientOptions)
const wsDuplexStream = WebSocket.createWebSocketStream(ws, { encoding: 'utf8' });
process.stdin.pipe(wsDuplexStream, { end: false })
process.stdin.on('end', () => wsDuplexStream.end(EOT))
wsDuplexStream.pipe(process.stdout)
```

- [pipe docs](https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options)